atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidatorTest.java

```diff
@@ -290,10 +309,37 @@ public void validateNewlyAddedHostsNewHostsAddedIfTheyAgreeWithOldHostsOnPreviou
                 .containsExactlyElementsOf(newCassandraServers);
     }
 
+    @Test
+    public void
+            validateNewlyAddedHostsNewHostsNotAddedIfTheyPresentAPossibleEvolutionOfPreviousTopologyAndCurrentServersExist() {
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = initialiseHosts(ALL_HOSTS);
+        Map<CassandraServer, CassandraClientPoolingContainer> oldHosts = EntryStream.of(allHosts)
+                .filterKeys(key -> OLD_HOSTS.contains(key.cassandraHostName()))
+                .toMap();
+        Map<CassandraServer, CassandraClientPoolingContainer> newHosts = EntryStream.of(allHosts)
+                .filterKeys(key -> NEW_HOSTS.contains(key.cassandraHostName()))
+                .toMap();
+        Set<CassandraServer> oldCassandraServers = filterServers(allHosts, OLD_HOSTS::contains);
+        Set<CassandraServer> newCassandraServers = filterServers(allHosts, NEW_HOSTS::contains);
+        Set<String> hostsOffline = ImmutableSet.of(OLD_HOST_ONE);
+        setHostIds(filterContainers(allHosts, hostsOffline::contains), HostIdResult.hardFailure());
+
+        setHostIds(filterContainers(newHosts, Predicate.not(hostsOffline::contains)), HostIdResult.success(UUIDS));
+        setHostIds(
+                filterContainers(oldHosts, Predicate.not(hostsOffline::contains)),
+                HostIdResult.success(MAYBE_SAME_AS_FIRST_CLUSTER));
+
+        validator.getNewHostsWithInconsistentTopologies(mapToTokenRangeOrigin(oldCassandraServers), oldHosts);
```

Comment:
Shouldn't we be asserting that this returns empty? That is, we actually did accept them as an evolution of previous topology?